src/log_prob.jl

```diff
@@ -0,0 +1,438 @@
+
+where the flux and position params are the same as the star case, and the
+shape params are unconstrained versions of (gdev, gaxis, gangle, gscale)
+
+Args:
+  - images: Vector of TiledImage types (data for log_likelihood)
+  - active_pixels: Vector of ActivePixels on which the log_likelihood is based
+  - ea: ElboArgs book keeping argument
+
+Returns:
+  - gal_logpdf  : unnormalized logpdf function handle that takes in a flat,
+                 unconstrained array as parameter
+  - gal_logprior: star param log prior function handle that takes in same
+                 flat, unconstrained array as parameter
+"""
+function make_galaxy_logpdf(images::Vector{TiledImage},
+                            active_pixels::Vector{ActivePixel},
+                            ea::ElboArgs)
+
+    # define star prior log probability density function
+    prior    = construct_prior()
+    subprior = prior.galaxy
+
+    function galaxy_logprior(state::Vector{Float64})
+        brightness, colors, u, gal_shape =
+            state[1], state[2:5], state[6:7], state[8:end]
+
+        # brightness prior
+        ll_b = color_logprior(brightness, colors, prior, true)
+        ll_s = shape_logprior(constrain_gal_shape(gal_shape), prior)
+        return ll_b + ll_s
+    end
+
+    # define star log joint probability density function
+    function galaxy_logpdf(state::Vector{Float64})
+        ll_prior = galaxy_logprior(state)
+
+        brightness, colors, position, gal_shape =
+            state[1], state[2:5], state[6:7], state[8:end]
+
+        ll_like  = state_log_likelihood(false, brightness, colors, position,
+                                        constrain_gal_shape(gal_shape), images,
+                                        active_pixels, ea)
+        return ll_like + ll_prior
+    end
+
+    return galaxy_logpdf, galaxy_logprior
+end
+
+
+"""
+Log likelihood of a single source given source params.
+
+Args:
+  - is_star: bool describing the type of source
+  - brightness: log r-band value
+  - colors: array of colors
+  - position: ra/dec of source
+  - gal_shape: vector of galaxy shape params (used if is_star=false)
+  - images: vector of TiledImage types (data for likelihood)
+  - active_pixels: vector of ActivePixels over which the ll is summed
+  - ea: ElboArgs object that maintains params for all sources
+
+Returns:
+  - result: a scalar describing the log likelihood of the
+            (brightness,colors,position,gal_shape) params conditioned on
+            the rest of the args
+"""
+function state_log_likelihood(is_star::Bool,
+                              brightness::Float64,
+                              colors::Vector{Float64},
+                              position::Vector{Float64},
+                              gal_shape::Vector{Float64},
+                              images::Vector{TiledImage},
+                              active_pixels::Vector{ActivePixel},
+                              ea::ElboArgs)
```

Comment:
Probably `ElboArgs` shouldn't appear in `log_prob.jl`, or really any mention of the ELBO. The `ElboDeriv` module (and the forthcoming `MCMC` module) imports `Model`, not the other way around.